### PR TITLE
Dev krushna

### PR DESCRIPTION
**Pull request by mardikark-gslab**
This pull request contains following changes
- Cleanup of "Column" references from DATA_LOAD operator
- LOAD_CSV changes to integrate with C code
- Introduced file_name() & nn_file_name() method in lVector and changes in reincarnate() method

### CODE REVIEW

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua [hunk 1]
```diff
@@ -162,7 +162,7 @@ return function (
           assert( status >= 0 , err.INVALID_DATA_ERROR )
         end   
            
-        column_list[col_idx+1]:put_chunk(1, cbuf, is_null)
+        column_list[col_idx+1]:put_chunk(cbuf, is_null, 1)
```
Comment (itsindra) [inline OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: quick question. Has the spec of put changed from length, chunk, nn_chunk to chunk, nn_chunk, length?

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua [hunk 2]
```diff
@@ -179,11 +179,12 @@ return function (
 
    for i, column in pairs(column_list) do
       column:eov()
+      print("EOV Done ", i)
```
Comment (itsindra) [inline OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: log this instead of printing it

File: OPERATORS/LOAD_CSV/lua/Makefile [hunk 3]
```diff
@@ -30,6 +30,7 @@ all : libload_csv.so
 ../gen_inc/_txt_to_SC.h : 
 	mkdir -p ../gen_inc/
 	lua ${UDIR}/cli_extract_func_decl.lua ../src/txt_to_SC.c ${INCDIR}
+
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/Makefile]: Why the extra newline

File: OPERATORS/LOAD_CSV/lua/Makefile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # For fast version, make -e DBGFLAGS="" -e OPTFLAGS=" -O4"
 #
-INCS= -I.  -I../../../UTILS/inc/ -I../gen_inc/  -I../../../UTILS/gen_inc 
+INCS= -I.  -I../../../UTILS/inc/ -I../gen_inc/  -I../inc/ -I../../../UTILS/gen_inc
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/Makefile]: is there an inc folder? I did not find it in the repo. If there is none, then lets not add the -I../inc/

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua [hunk 5]
```diff
@@ -179,11 +179,12 @@ return function (
...
       -- print(column:length())
       -- if no nulls are present, then delete the null file
       if ( ( M[i].has_nulls ) and ( M[i].num_nulls == 0 ) ) then
-        local null_file = require('Q/q_export').Q_DATA_DIR .. "/_" .. M[i].name .. "_nn"
-        assert(plfile.delete(null_file),err.INPUT_FILE_NOT_FOUND)
+        --local null_file = require('Q/q_export').Q_DATA_DIR .. "/_" .. M[i].name .. "_nn"
+        --assert(plfile.delete(null_file),err.INPUT_FILE_NOT_FOUND)
```
Comment (subramon) [inline OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: What did we decide about deleting nn files that are not needed?

File: OPERATORS/F1S1OPF2/lua/f1s1opf2.lua [hunk 6]
```diff
@@ -1,7 +1,7 @@
 local T = {}
```
Comment (subramon) [inline OPERATORS/F1S1OPF2/lua/f1s1opf2.lua]: Actually, I have deleted thsi file.
We use _f1s1opf2.lua which is generated by pkg_f1s1opf2.lua

File: UTILS/src/f_mmap.c [hunk 7]
```diff
@@ -22,7 +22,7 @@
 
 MMAP_REC_TYPE *
 f_mmap(
-   char * const file_name,
+   const char * const file_name,
```
Comment (subramon) [inline UTILS/src/f_mmap.c]: I think we need to delete the first const. Let's discuss

File: OPERATORS/LOAD_CSV/lua/load_csv.lua [hunk 8]
```diff
@@ -155,123 +155,193 @@ load_csv = function (
...
-  local x_idx = 0
-  local in_buf  = ffi.malloc(max_txt_width)
-  local row_idx = 1
-  local col_idx = 1
-  local num_in_out_buf = 0
-  local num_cols = #M
+  -- checking if there are any SC SV qtype cols
+  local is_SC_SV_present = false
+  
+  local no_of_cols = #M
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/load_csv.lua]: why create a separate a variable for this?

File: OPERATORS/LOAD_CSV/lua/load_csv.lua [hunk 9]
```diff
@@ -155,123 +155,193 @@ load_csv = function (
...
+    for i = 1, #M do
+      fldtypes[i-1] = ffi.malloc(4 * ffi.sizeof("char"))
+      fldtypes[i-1] = ffi.cast("char *", fldtypes[i-1])
+      ffi.copy(fldtypes[i-1], M[i].qtype)
+    end 
+    
+    local is_hdr = false
+    
+    local is_load = ffi.malloc(#M * ffi.sizeof("bool"))
+    is_load = ffi.cast("bool *", is_load)
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/load_csv.lua]: combine the two loops

File: OPERATORS/LOAD_CSV/lua/load_csv.lua [hunk 10]
```diff
@@ -155,123 +155,193 @@ load_csv = function (
...
+      is_hdr, is_load, has_nulls, num_nulls, out_files, nil_files,
+      str_for_lua, sz_str_for_lua)  
+    assert(status == 0, "load_csv_fast failed")
+    
+    local vector_string = ffi.string(ffi.cast("char *",str_for_lua))
+    assert(( vector_string ) and ( vector_string ~= "" ), "load_csv_fast returned a null string" )
+    
+    loadstring(vector_string)()
+    
+    assert( (type(T) == "table" and type(T[1]) == "lVector" ), "type of T is not lVector")
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/load_csv.lua]: Break this up into smaller functions, This is too big a function

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua (hunk 1 above)
Comment (mardikark-gslab) [reply OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: Yes, lVector specification for put_chunk is as below
lVector:put_chunk(base_addr, nn_addr, len)

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua (hunk 2 above)
Comment (mardikark-gslab) [reply OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: This print statement was added while debugging, it is not required as such, removing it

File: OPERATORS/LOAD_CSV/lua/Makefile (hunk 4 above)
Comment (mardikark-gslab) [reply OPERATORS/LOAD_CSV/lua/Makefile]: Yes, there is a inc directory in LOAD_CSV operator directory containing 'load_csv_fast.h'.
Location: Q/OPERATORS/LOAD_CSV/inc/load_csv_fast.h

File: OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua (hunk 5 above)
Comment (mardikark-gslab) [reply OPERATORS/DATA_LOAD/lua/load_csv_dataload.lua]: We decided to provide a API in lVector level to delete the nn file associated with the vector which will be a cleaner approach.

File: OPERATORS/F1S1OPF2/lua/f1s1opf2.lua (hunk 6 above)
Comment (mardikark-gslab) [reply OPERATORS/F1S1OPF2/lua/f1s1opf2.lua]: Yes, it was not getting used but it was there so I thought to modify it

File: UTILS/src/f_mmap.c (hunk 7 above)
Comment (mardikark-gslab) [reply UTILS/src/f_mmap.c]: Ok

File: OPERATORS/LOAD_CSV/lua/load_csv.lua (hunk 10 above)
Comment (mardikark-gslab) [reply OPERATORS/LOAD_CSV/lua/load_csv.lua]: Ok

**Pull request merged**